### PR TITLE
Allow mixed case and fix exceptions in from_hex

### LINKED ISSUE #2669: Allow mixed case in from base functions
For example:
```
com.google.common.io.BaseEncoding$DecodingException: Unrecognized character: a
java.lang.IllegalArgumentException
com.google.common.io.BaseEncoding.decode(BaseEncoding.java:228)
com.facebook.presto.operator.scalar.VarbinaryFunctions.fromHexVarchar(VarbinaryFunctions.java:98)
```

Also invalid input should be classified as `INVALID_FUNCTION_ARGUMENT`

### PR DESCRIPTION
**Pull request by haozhun**
Fixes #2669

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/VarbinaryFunctions.java [hunk 1]
```diff
@@ -95,7 +98,29 @@ public static Slice toHex(@SqlType(StandardTypes.VARBINARY) Slice slice)
...
+
+    private static int hexDigitCharToInt(byte b)
+    {
+        if (b >= '0' && b <= '9') {
+            return b - '0';
+        }
+        else if (b >= 'a' && b <= 'f') {
+            return b - 'a' + 10;
+        }
+        else if (b >= 'A' && b <= 'f') {
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/VarbinaryFunctions.java]: This should be a capital 'F'

Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/VarbinaryFunctions.java]: Add a test that would have caught this too

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/VarbinaryFunctions.java [hunk 2]
```diff
@@ -95,7 +98,29 @@ public static Slice toHex(@SqlType(StandardTypes.VARBINARY) Slice slice)
...
+        byte[] result = new byte[slice.length() / 2];
+        for (int i = 0; i < slice.length(); i += 2) {
+            result[i / 2] = (byte) (hexDigitCharToInt(slice.getByte(i)) * 16 + hexDigitCharToInt(slice.getByte(i + 1)));
+        }
+        return Slices.wrappedBuffer(result);
+    }
+
+    private static int hexDigitCharToInt(byte b)
+    {
+        if (b >= '0' && b <= '9') {
```
Comment (yaoxin226) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/VarbinaryFunctions.java]: How about use Character.isDigit(b), Character.isLowerCase(b), Character.isUpperCase(b)

Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/VarbinaryFunctions.java]: :+1:

Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/VarbinaryFunctions.java]: `isDigit(b)` is way more complicated than `b>='0' && b <= '9'`. For example, ８, ໘, ९, ٣ will give you `true`. You will pay the price even if you don't care. And if you care, this becomes even worse.

**Comment by haozhun** [comment]
Fixed comments. @dain please take a look since you filed the original issue.

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/VarbinaryFunctions.java [hunk 3]
```diff
@@ -95,7 +98,29 @@ public static Slice toHex(@SqlType(StandardTypes.VARBINARY) Slice slice)
...
     public static Slice fromHexVarchar(@SqlType(StandardTypes.VARCHAR) Slice slice)
     {
-        return Slices.wrappedBuffer(BaseEncoding.base16().decode(slice.toStringUtf8()));
+        if (slice.length() % 2 != 0) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "invalid input length " + slice.length());
+        }
+
+        byte[] result = new byte[slice.length() / 2];
+        for (int i = 0; i < slice.length(); i += 2) {
+            result[i / 2] = (byte) (hexDigitCharToInt(slice.getByte(i)) * 16 + hexDigitCharToInt(slice.getByte(i + 1)));
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/VarbinaryFunctions.java]: use `<< 4` and `&` instead of `*` and `+`

**Comment by dain** [comment]
Looks good

**Pull request closed without merging**